### PR TITLE
add test for nats timeouts

### PR DESCRIPTION
**Pull request by timkaye11**
creates a test for testing NATS timeouts: 
 - send messages  to a no-op receiver until we get an error

### CODE REVIEW

File: conn_test.go [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+	defer receiver.Close()
+
+	done := make(chan bool)
+	expected := errFailedToSend(defaultSendRetries, defaultSendRetries*timeout)
+	go func() {
+		defer close(done)
+		for {
+			if err := sender.Send("receiver", []byte("lol")); err != nil {
+				if expected.Error() != err.Error() {
+					t.Fatalf("unexpected error: %v", err)
```
Comment (mdmarek) [inline conn_test.go]: Can't user `t` in another go-routine, documented here: https://golang.org/pkg/testing/#B.FailNow

> FailNow must be called from the goroutine running the test or benchmark function, not from other goroutines created during the test.

File: conn_test.go [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+				return
+			}
+		}
+	}()
+
+	select {
+	case <-time.After(2 * defaultSendRetries * timeout):
+		t.Fatalf("test timed out")
+	case <-done:
+	}
```
Comment (mdmarek) [inline conn_test.go]: You could add another channel here, with size 1, where the go-routine can send a signal if it enters the if-block checking the actual error to the expected error.

Or something like that :)

Comment (timkaye11) [inline conn_test.go]: whoops, fixed :)

File: conn_test.go [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+	timeout := time.Second
+	SetConnSendTimeout(sender, timeout)
+
+	receiver, err := NewReceiver(ec, "receiver", 1, 0)
+	if err != nil {
+		t.Fatalf("error creating rx: %v", err)
+	}
+	defer receiver.Close()
+
+	errs := make(chan error)
```
Comment (mdmarek) [inline conn_test.go]: Probably just move the `defer close(errs)` here.

File: conn_test.go [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+	expected := errFailedToSend(defaultSendRetries, defaultSendRetries*timeout)
+	go func() {
+		for {
```
Comment (mdmarek) [inline conn_test.go]: Does this need the for-loop?

File: conn_test.go (hunk 3 above)
Comment (timkaye11) [inline conn_test.go]: fixed

**Comment by mdmarek** [comment]
:+1:

**Pull request merged**